pkg/scheduler/internal/cache/snapshot_test.go

```diff
@@ -229,3 +230,251 @@ func TestCreateUsedPVCSet(t *testing.T) {
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1"},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-2", Labels: map[string]string{"baz": "qux"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNodesInfos: []*framework.NodeInfo{
+				{
+					Pods: []*framework.PodInfo{},
+				},
+				{
+					Pods: []*framework.PodInfo{
+						{
+							Pod: podWithAnnotations,
+						},
+					},
+				},
+				{
+					Pods: []*framework.PodInfo{
+						{
+							Pod: podsWithAffitiny[0],
+							RequiredAffinityTerms: []framework.AffinityTerm{
+								{
+									Namespaces:        sets.NewString("ns"),
+									Selector:          labels.SelectorFromSet(map[string]string{"baz": "qux"}),
+									TopologyKey:       "baz",
+									NamespaceSelector: labels.Nothing(),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedNumNodes:         3,
+			expectedPodsWithAffinity: 1,
+		},
+		{
+			name: "multiple nodes, pod with affinity, pod with anti-affinity",
+			pods: []*v1.Pod{
+				podsWithAffitiny[1],
+				podWithAntiAffitiny,
+			},
+			nodes: []*v1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-0", Labels: map[string]string{"key": "value"}},
+					Status:     v1.NodeStatus{},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node-1", Labels: map[string]string{"another": "label"}},
+					Status:     v1.NodeStatus{},
+				},
+			},
+			expectedNodesInfos: []*framework.NodeInfo{
+				{
+					Pods: []*framework.PodInfo{
+						{
+							Pod: podsWithAffitiny[1],
+							RequiredAffinityTerms: []framework.AffinityTerm{
+								{
+									Namespaces:        sets.NewString("ns"),
+									Selector:          labels.SelectorFromSet(map[string]string{"key": "value"}),
+									TopologyKey:       "key",
+									NamespaceSelector: labels.Nothing(),
+								},
+							},
+						},
+						{
+							Pod: podWithAntiAffitiny,
+							RequiredAntiAffinityTerms: []framework.AffinityTerm{
+								{
+									Namespaces:        sets.NewString("ns"),
+									Selector:          labels.SelectorFromSet(map[string]string{"another": "label"}),
+									TopologyKey:       "another",
+									NamespaceSelector: labels.Nothing(),
+								},
+							},
+						},
+					},
+				},
+				{
+					Pods: []*framework.PodInfo{},
+				},
+			},
+			expectedNumNodes:             2,
+			expectedPodsWithAffinity:     1,
+			expectedPodsWithAntiAffinity: 1,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			snapshot := NewSnapshot(test.pods, test.nodes)
+
+			if test.expectedNumNodes != snapshot.NumNodes() {
+				t.Errorf("unexpected number of nodes, want: %v, got: %v", test.expectedNumNodes, snapshot.NumNodes())
+			}
+
+			for i, node := range test.nodes {
+				info, err := snapshot.Get(node.Name)
+				if err != nil {
```

Comment:
Every `err != nil` is now replaced with three lines, similar with `info == nil` below. Do we have a package to do those checks in a single line?

Reply:
If this is the case please tell me and I will update the PR

Reply:
Also not sure about this package. LGTM it's easy to read.